libr/crypto/p/crypto_sm4.c

```diff
@@ -0,0 +1,193 @@
+/* *
+ * radare - LGPL - Copyright 2017-2022 - Sylvain Pelissier
+ * Implementation of SM4 block cipher
+ * https://datatracker.ietf.org/doc/html/draft-ribose-cfrg-sm4-10
+ * 
+ * */
+
+#include <r_lib.h>
+#include <r_crypto.h>
+#include <memory.h>
+
+#define BLOCK_SIZE 16
+#define SM4_KEY_SIZE 16
+
+
+/* Round keys */
+ut32 sk[32];
```

Comment:
We dont like global variables but if theres no escape. Make it "static R_TH_LOCAL"